smart_open/utils.py

```diff
@@ -223,3 +223,10 @@ def __exit__(self, *args, **kwargs):
 
     def __next__(self):
         return self.__wrapped__.__next__()
+
+    def close(self):
+        try:
+            self.__wrapped__.close()
+        finally:
+            if self.__inner != self.__wrapped__:  # Don't close again if inner and wrapped are the same
```

Comment:
I don't think there is a scenario where they are the same, plus calling close a second time should be a no-op. I think we can do without this conditional (unless there are tests failing without this conditional?)

Reply:
Thanks for checking the code @ddelange . This following test fails if I remove this conditional:

https://github.com/jbarragan-bridge/smart_open/blob/f435238348ce6ab66bc2cc8a3ab13a5a4595c145/smart_open/tests/test_smart_open.py#L1409-L1445 :
```python 
assert len(responses.calls) == 4  # it returns 6 without the conditional
```

I don't know much about `hdfs`; I saw in debug mode that this scenario creates a `FileLikeProxy` instance where the `wrapper` and `inner` are the same; duplicating the `close()` call.

Should adjust the unit test instead?

Reply:
good to know, something to look into but not related to this PR. let's leave it as is :)